### ISSUE TITLE
l10n_cn

### ISSUE THREAD

**Issue by lap-odoo** [odoo/odoo#18512]
Dear Odoo Partners,

At Odoo, a lot of effort is put into improving the accounting localizations for each country and also to maintain these accounting localizations up-to-date. To achieve that, we will need your help.

One github issue will be created per accounting localization (e.g. l10n_be for Belgium). We expect from you to make suggestions about what should be improved in the specific accounting localization. Do not hesitate to share files with detailed explanations.

As soon as changes will be implemented, you will be informed and will have the opportunity to test it and share your feedback.

This github issue will also be useful to keep you informed about all the ongoing improvements concerning a specific accounting localization.

**Comment by lap-odoo** [comment]
Dear Partners, 

We have already started to make some improvements for the Chinese Localization. 

Regarding the taxes, we will add some to have: 
- Sales, included : 3%, 6%, 11%, 17%
- Sales, excluded : 3%, 6%, 11%, 17%
- Purchases, excluded : 3%, 6%, 11%, 17%

The default taxes will be:
- Sales: 17% Incl. 
- Purchase: 17% Excl. 

Could you please confirm this is correct and complete? 

Are there fiscal positions needed? 

Regarding the CoA, we will keep the Standard CoA and the Small Businesses CoA. We will improve them and make sure we have correct translations (ZH-EN). If you have suggestions, do not hesitate to send them to us. 

We also would like to update the exchange rates automatically based on the Bank of China. If you have information, that would be helpful. 

Thanks in advance

**Comment by elicoidal** [comment]
A good start for BoC rate could be here: http://www.boc.cn/sourcedb/whpj/enindex.html

**Comment by ed29** [comment]
Yes,usually,the taxes are 3%, 6%, 11%, 17%
For more,the taxes are 0%, 3%, 5%, 6%, 11%, 13%, 17%, refer to http://www.chinaacc.com/zhuanti/cssw/hu1604152689.shtml

**Comment by ed29** [comment]
+1
update the exchange rates automatically

**Comment by jeffery9** [comment]
yes,  I should do some works for those.

**Comment by chinamfg** [comment]
I'd liket to do something!

**Comment by lvx5013** [comment]
Cost Accounting and Cost Keeping compliance would be the most urgent issue we would like to address. and maybe a good place to start in terms of Chinese localization.

Warmest,
Li

**Comment by lap-odoo** [comment]
@ed29 : Thanks for the info regarding the taxes. Any idea where we could find detailed information about that in English? 

@lvx5013 : Could you please provide more info about what you mean with cost accounting and cost keeping? Thank you

**Comment by ed29** [comment]
@lap-odoo Yes,in english is best,but sorry for that i don't find infomation in english,maybe google translate would help,that are texts in the page.

**Comment by lvx5013** [comment]
First of all,

Chinese accrual basis:

Accrual basis (accrual basis), also known as accrual basis, refers to the right to receive cash or to pay cash, responsibility, the occurrence of power and responsibility as a sign to confirm the current income and expenses, and claims and liabilities.

That is, revenues are recognised as cash receipts and future cash receipts, claims, and expenses are recognised as cash disbursements and future cash disbursements - the occurrence of debt. Instead of confirming the income and expenses with cash receipts and payments.

**Comment by lvx5013** [comment]
According to the accrual principle, all the existing income and has occurred or shall bear the costs, regardless of whether the money has been cash income and expenses should be as treatment period; all income and expenses not attributable to the current period, even if the money had been received pay in the current period, should not be recognized as income and expenses. Therefore, accrual basis is the requirement of accounting element confirmation and measurement. It solves the problem of when and how much revenue and expenses should be confirmed and confirmed.


current income

Any income that has already been realized in the current period, whether or not received funds (money), should be recorded on the account of the current revenue increase. If income is not included in the current period, even if the amount (money) has been received, the accountant can not record the current income increase on the books.

Current expenses

Accounting for any expenses that have occurred or should be incurred in the current period, whether or not they have paid the money (money), the accounts should be recorded on the books and the cost of the current period has increased. If the amount of money (money) has already been paid, the accountant can not record the cost of the current account.

**Comment by lvx5013** [comment]
Fixed assets purchase valuation

- China stipulates that fixed assets are usually priced at their historical cost or net value. Change the depreciation method as an accounting policy change. 

- According to international accounting standards, fixed asset pricing should consider the effects of fair value, including revaluation and discounting. Change depreciation method as a change in accounting estimates. Suppose an enterprise January 1, 2003 purchase, the price of 100 million yuan, 10 years matching repayment, the current cash flow value of 90 million yuan.

How Borrowing costs are capitalized. 

- China's guidelines specify that borrowing costs incurred for the purchase of a fixed asset shall be capitalized in the event of capitalization, until the asset reaches a predetermined state of use. 
Capitalized amount = loan interest + auxiliary cost and exchange balance. 

The international accounting standard provides that general borrowing for the construction of assets meets the requirements and can be capitalized. Capitalization amount = all special borrowing costs - temporary investment income. 

- Suppose that an enterprise borrowed 10 million yuan in January 1, 2003, and specialized loans were used to build a factory building. The average annual expenditure of 03 yuan was 8 million, and the balance of the amount borrowed by the enterprise will be placed in the special bank account, and the interest income will be 50 thousand. 

- The capitalization amount calculated under Chinese standards = 800 thousand, according to the international accounting standards, the amount of capitalization = 950 thousand. In this way, IAS amounts to more than 150 thousand of capitalization and less recognition of interest expense 150 thousand than PRC. 

- Non monetary easy accounting standards in China, non monetary exchange of assets into the assets of the book value of the relevant taxes and fees. 

- The international accounting standards to distinguish between similar and non similar provisions of the non monetary asset swap, easy to change into the similar assets according to the exchange value of assets and related taxes accounted for, and not the same exchange of assets at fair value from assets, profit and loss = change the carrying value of the asset exchange into the fair value of assets.

**Comment by lvx5013** [comment]
Short-term investment

China's guidelines specify that short-term investments are measured at the lower of cost and market price, and are only included in profit or loss for loss of value.
International accounting standards provide that short-term investments are measured at fair value and are subject to profit or loss changes in market and cost.

Long term investment goodwill

- China standards of equity investment in the investment cost, investment cost, credits included in capital reserve.
- The international accounting standard provides that investments in debt securities held to maturity date are weighted by the weighted average of the recognized depreciation / amortization assets at the cost
- Remaining years amortization, such as negative goodwill over FV, more than immediately recognized as income.
- Suppose the enterprise A uses 1 million yuan to buy the enterprise B30%'s stock right, the enterprise B net asset is 2 million yuan, but the net assets fair value total is 3 million
- Yuan. The difference in equity investment calculated under Chinese standards is 400 thousand, and goodwill is calculated according to international accounting standards = 100 thousand, and thereafter shares are invested

Profits and losses, the need to consider the impact of adjustment due to fair value.

Intangible assets research and development costs

- China's guidelines specify that research and development costs are directly included in the current profits and losses, and shall not be capitalized, except for the development of intangible assets for the purpose of obtaining the application according to law Production, such as registration fees, lawyers and other expenses can be capitalized.
- International accounting standards specify that research expenditures are recognised as expenses at the time of their occurrence, and that they are recognised as intangible assets when they meet certain requirements.

Organization expenses

- China standard stipulates that the start-up expenses shall be collected in the long-term expenses to be amortized, and shall be included in the month after the beginning of the production and operation of the enterprise until the beginning of the month Gains and losses.
- International accounting standards provide that the start-up expenses are recognized as expenses during the current period.

Asset class government grants

- According to the standards of China, enterprises shall, according to the quota of subsidies provided by the state, calculate the fixed subsidies according to the subsidy quotas stipulated by the state, and shall be paid at the end of the term. The amount of subsidy receivable shall be credited to the allowance income.
- International accounting standards provide for asset related government grants as deferred income and are systematically and reasonably recognized as proceeds in the asset's useful life.

**Comment by liuqinfalinwa** [comment]
@lap-odoo 
Refer to tax system from the government(updated in 2015)
[](http://www.chinatax.gov.cn/eng/n2367731/index.html)
you can check all the tax used in China.

Based on that link（update in 2015）, we have some reforms in 2016 and 2017
VAT Pilot Reform in 2016: Business Tax to VAT Rolled Out Comprehensively, after that selling goods and services in China, we mainly use VAT and no more business tax.
VAT Reform in 2017(up to August 2017): cancel VAT rate 13%.

so purchase and sales VAT we use now is:
3%, 6%, 11%, 17%

**Comment by lap-odoo** [comment]
Hi All, 

We have tried to regroup the information we received and what we already have in Odoo to build a more complete CoA. We end up with the result attached. What do you think of it ? Is it complete enough ? If yes, we would need help for the translations, account type, ... µ

Thanks
[import_odoo_accounting_chinese.zip](https://github.com/odoo/odoo/files/1207918/import_odoo_accounting_chinese.zip)

**Comment by lvx5013** [comment]
Translated Version, working on account types now.


[import_odoo_accounting_chinese_translated_v_0_9.zip](https://github.com/odoo/odoo/files/1227304/import_odoo_accounting_chinese_translated_v_0_9.zip)

**Comment by lvx5013** [comment]
**Chinese accrual basis accounting**

Question 1: A sales order has been created, but goods not yet shipped, should we count this as Income?
Answer: No. Only when shipment has been made, customer has received this shipment, this amount can be counted as income.

Question 2: Received cash for a sales order, but shipment hasn't been made, should this be counted into  Income?
Answer: No. Only when shipment has been made, customer has received the shipment, this amount can be counted as Income.

**Comment by lvx5013** [comment]
@lap-odoo 

**Problem 1:** Odoo Accounting types is different from Chinese Standard accounting types.
Financial reports in Odoo rely on Accounting types to determine the sum. 

Therefore: Standard Chinese Chart of Account needs to be tweaked to obtain a correct financial report in Odoo.

* Some of our Chart of Account are forced to be marked as DO NOT REFLECT ON FINANCIAL REPORT category. In order to have a correct Income Statement / Balance Sheet / etc in Odoo.

**Problem 2:** Chinese Legal Financial Reports rely on accounting types to sum the data, however, this won't work in Odoo since some of our Chart of Account weren't used due to Problem 1. 


**Solution Proposal:**

- 1. Add a separated entry in Chart of Account, Chinese Accounting Types.
- 2. Add a separated Chinese Accounting Reports view in Odoo's standard financial reports.


After this, everything will smooth out accordingly.

Warmest,
Li

**Comment by lap-odoo** [comment]
@lvx5013 : 

We've discussed Chinese accrual basis accounting internally. 

Question 1: A sales order has been created, but goods not yet shipped, should we count this as Income? Answer: No. Only when shipment has been made, customer has received this shipment, this amount can be counted as income.
> This should be managed with a good invoicing policy. The invoice should only be created once the delivery is done. Odoo can already help you to do that, you just have to choose "based on delivered quantities" as invoicing policy on your products. 

Question 2: Received cash for a sales order, but shipment hasn't been made, should this be counted into Income? Answer: No. Only when shipment has been made, customer has received the shipment, this amount can be counted as Income.
> The income account is only impacted at the invoice creation. So if you create the invoice after the delivery (invoicing policy based on delivered quantities), there is no issue.

**Comment by lap-odoo** [comment]
@lvx5013 : Thank you for your answer about the CoA. 

If the issue is that some accounts shouldn't appear in the reports (Balance Sheet, P&L, ...), we can create a new account type for that. 

Would that be enough? 

If you don't want to use the Standard P&L and BS from Odoo (GAAP reports), we can create reports specific for China. In that case, these reports should be based on the tags defined on the accounts (not on their types). 

Thanks for your help

**Comment by lvx5013** [comment]
@lap-odoo 

[Chinese_Chart_of_Account_2017.zip](https://github.com/odoo/odoo/files/1240725/Chinese_Chart_of_Account_2017.zip)

Wonderful! 

I have attached a Top Rated Government accepted SME CoA lay into the six Chinese Government Accredited account types:

Type 1: Assets;
Type 2: Debts;
Type 3: Common;
Type 4: Owner's Equity;
Type 5: Cost;
Type 6 a:   Income;
Type 6 b:  Expense;

Cheers! Look forward to our final product!

**Comment by lvx5013** [comment]
@lap-odoo 

A standard Chinese SME income statement is attached (used for Quarterly Government Financial Declaration in China), which differs from the GAAP financial reports. Thank you a lot! Look forward into it!  

[Chinese_Income_Statement.pdf](https://github.com/odoo/odoo/files/1240774/Chinese_Income_Statement.pdf)
[Chart_of_Account_2017_English_Income_Statement.pdf](https://github.com/odoo/odoo/files/1240792/Chart_of_Account_2017_English_Income_Statement.pdf)

**Comment by lap-odoo** [comment]
@lvx5013 I'm a bit lost. You also sent me this CoA. 

[2017最新会计科目及代码表_2017_Chinese_accounting_subjects (1).zip](https://github.com/odoo/odoo/files/1241784/2017._2017_Chinese_accounting_subjects.1.zip)

I know there are two CoA for l10n_CN, the standard one and the one for the SMEs. 

Concretely, what we need now is one csv for Standard CoA and one csv for SME CoA. In these csv, we need : 
- The account code
- The account name (+ translation in English) 
- The account type (you must only use the account types that already exist in Odoo, if needed we can create an account type "do not show in reports" as discussed) 
- The  account tags (you can create the ones you need. You can give them the name of the Chinese Account Types for example. The Chinese standard legal reports will be based on these tags, and not on the account types)

Once we have this, we will be able to create BS and P&L specific to China (based on the tags of the accounts). 

I hope it's clearer. 

Thanks a lot

**Comment by lvx5013** [comment]
[中国会计-中小企业-CoA.zip](https://github.com/odoo/odoo/files/1247875/-.-CoA.zip)

Great!

I've attached the SME-CoA with:

- The account code
- The account name (+ translation in English)
- The account type (the account types that already exist in Odoo)
- The account tags (Chinese Account Types)

Cheers

**Comment by lvx5013** [comment]
@lap-odoo 
In the CoA above, Account No.4001 to 4104 should be categorized into owner's equity.  Sorry for the inconvenience.

Best
[中小企业CoA2017829.zip](https://github.com/odoo/odoo/files/1258980/CoA2017829.zip)

Li

**Comment by lap-odoo** [comment]
@lvx5013 Thank you ! 

There are much less accounts than in the current Odoo CoA. 

Could the other partners confirm me this CoA is better than the current one ? Thank you

**Comment by lukebranch** [comment]
@elicoidal ,

I've heard some rumours that the accounting software used for reporting to the China government must be certified by the China government. I have not been able to verify this as fact/fiction at this point in time though.

Any idea if this is true or not? I assume as long as the software complies with China accounting standards it will be acceptable.

**Comment by elicoidal** [comment]
@lukebranch 
YonYou/Kingdee/SAP are supposed to be the happy few with domestic certifications (there might be others but not that many). Process for tax bureau certification is tedious: you need to parallel run for a while, demonstrate that the software is "safe" (difficult in the opensource) etc. I have studied it time ago and came to the conclusion it was unreachable at the time for Odoo (at least as a partner).
 
Still, I am in the same state as you are and cannot really confirm/deny the extent of the "rumor". We have some customers only using Odoo for their tax declaration: it will mainly depend on tax bureau and the probably that the tax bureau ask for an audit of your accounting software (which is nearly 0 for Chinese SMB). Bigger WFOE companies might have higher probability but honestly, unless you need yearly internal audit (not mandatory for Chinese SMB), it is unlikely.
I came to the conclusions over the years that basically, using similar reports as Kingdee/Yonyou will just please everybody and avoid raising questions without any risk.

I have planned to work with an accountant in the coming weeks to get a better understanding. Maybe I come back on that later

**Comment by lukebranch** [comment]
@elicoidal ,

> YonYou/Kingdee/SAP are supposed to be the happy few with domestic certifications (there might be others but not that many). Process for tax bureau certification is tedious: you need to parallel run for a while, demonstrate that the software is "safe" (difficult in the opensource) etc. I have studied it time ago and came to the conclusion it was unreachable at the time for Odoo (at least as a partner).

> I came to the conclusions over the years that basically, using similar reports as Kingdee/Yonyou will just please everybody and avoid raising questions without any risk.

Thanks for the invaluable feedback. We are thinking along the same lines (design the report layout to mimic the output from Kingdee in our case).

It sounds like getting certified is something that Odoo SA/HK should seriously consider though, as it appears it would give Odoo (as a platform) some considerable leverage within the ERP market in China.

> Still, I am in the same state as you are and cannot really confirm/deny the extent of the "rumor". We have some customers only using Odoo for their tax declaration: it will mainly depend on tax bureau and the probably that the tax bureau ask for an audit of your accounting software (which is nearly 0 for Chinese SMB). Bigger WFOE companies might have higher probability but honestly, unless you need yearly internal audit (not mandatory for Chinese SMB), it is unlikely. 
> 
> I have planned to work with an accountant in the coming weeks to get a better understanding. Maybe I come back on that later
> 

I'd definitely be interested in what they have to say. As of 11.0 with some of the new financial reporting features, it appears it might be far more feasible to achieve. We are in the process of the initial stages of implementation now. If we discover anything during this process we'll certainly update this thread as well.

@lap-odoo @Yenthe666 , is this something Odoo SA/HK has a roadmap for in terms of targeting official certification for a l110_cn* localisation for Odoo 12.0? It seems like it would be worthwhile to look into this.

**Comment by lap-odoo** [comment]
@elicoidal Thanks for the information! 

@lukebranch No, we don't plan to do it in the short term. As explained by @elicoidal , it's still quite unclear if this certification is really needed and what needs to be done to obtain it. We may want to have a look at it in the future but we first want to improve the financial reports and the general features of the localizations.

**Comment by lukebranch** [comment]
FWIW, i've come across these regulations from the PRC as provided by a customer currently using Kingdee:

> l  Here is the Accounting files management regulation of PRC for your reference: http://tfs.mof.gov.cn/zhengwuxinxi/caizhengbuling/201512/t20151214_1613338.html
>  
> l  Here is the Corporation Informatization accounting work regulation of PRC for your reference, the regulation stated the requirement of the software be used for company. http://www.mof.gov.cn/zhengwuxinxi/caizhengwengao/wg2013/wg201312/201406/t20140609_1092515.html
>  
> l  Here is the Accounting law of PRC for your reference: http://www.mof.gov.cn/zhuantihuigu/kjfzth20/bjzl/200805/t20080519_20769.html

Google translate does a pretty good job of getting the requirements across.

This whole document goes into detail about the legal requirements from the PRC government regarding IT Accounting system requirements:
http://www.mof.gov.cn/zhengwuxinxi/caizhengwengao/wg2013/wg201312/201406/t20140609_1092515.html

Here are a few excerpts translated into English:

> Article 10:
> The accounting software shall provide the irreversible accounting function to ensure the continuous serial numbering of the similar vouchers that have been booked and shall not provide the function of deleting and inserting the vouchers already booked. No provision shall be made for the date, amount, subjects and Operator's modification function.
> 
> Article 11:
> Article 11 Software vendors are encouraged to integrate Extensible Business Reporting Language (XBRL) capabilities into their accounting software so that enterprises can generate XBRL financial reports that conform to the national uniform standards.
> 
> (XBRL, or eXtensible Business Reporting Language, is an XML standard for tagging business and financial reports to increase the transparency and accessibility of business information by using a uniform format. - https://www.xbrl.org/the-standard/what/an-introduction-to-xbrl/)
> 
> Article 44:
> Where an enterprise uses accounting software that does not meet the requirements of this code, the financial department shall order it to make corrections within a prescribed time limit. If the deadline is not changed, the public finance department shall make a public announcement and notify the relevant department at the same level or its dispatched institution of the relevant information.
> 
> Article 49:
> These Rules shall come into force on January 6, 2014, the Rules for the Review of Commercial Accounting Software (Cai Shui Zi [1994] No. 27) promulgated by the Ministry of Finance on June 30, 1994, the Accounting and Computing Management Measures "(Accounting Association 〔1994〕 27) at the same time abolished.

http://www.mof.gov.cn/zhuantihuigu/kjfzth20/bjzl/200805/t20080519_20769.html

> 
>    Article 26 Companies and enterprises shall not conduct any of the following acts in accounting
>     (1) To arbitrarily change the recognition standard or measurement method of assets, liabilities and owner's rights and interests, assets, liabilities and owners'
>     (B) virtual or conceal revenue, postpone or confirm revenue ahead of schedule;
>     (3) To arbitrarily change the criteria for determining the cost or cost, or to calculate the cost, cost, cost;
>     (4) To arbitrarily adjust the calculation and distribution methods of profits, fabricate false profits or conceal profits;
>     (5) Other acts in violation of the unified national accounting system.

**Comment by lap-odoo** [comment]
@lukebranch Thanks ! 

We have already worked on a module to ensure that the accounting entries are unalterable (for France), I think we can adapt it for China too, I'll add it in the backlog. As this is mandatory, I think it should be the priority (with a correct CoA and correct taxes).

**Comment by jeffery9** [comment]
tax changed.  17% change to 16%, both sales and puchase tax

**Comment by Guobower** [comment]
The fundamental problem with Odoo localization is that when the language template is generated, the contents of the Html tag are incorrectly included, and a series of translation problems are generated.

**Comment by Guobower** [comment]
Financial localization, first of all to solve the financial provisional assessment problem, by tracking whether the supplier invoice is received, to process the corresponding warehousing vouchers, then the financial monthly carry-over, and generate a financial statement with the previous period balance

**Comment by C3POdoo** [comment]
Dear @lap-odoo,

Thank you for your report but we are closing it due to inactivity.
We apology if we could not look at your request in time.
If your report still makes sense, don't hesitate to reopen a new one. We will try to check it as soon as possible.

_This is an automated message._
